src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -144,6 +143,21 @@ QgsAdvancedDigitizingDockWidget::QgsAdvancedDigitizingDockWidget( QgsMapCanvas *
   {
     commonAngles << QPair<double, QString>( *it, formatCommonAngleSnapping( *it ) );
   }
+
+  {
+    QAction *action = new QAction( tr( "Snapping to features has priority over common angles" ), mCommonAngleActionsMenu );
```

Comment:
This is rather cumbersome action text. How about this approach instead?

We add a submenu "Snapping Priority", and it has mutually exclusive actions in it for "Prioritise Snapping to Features" and "Prioritise Snapping to Angle". (Using QActionGroup to make the actions exclusive). 

IMO that's a bit more user friendly and avoids the long text.

Reply:
done